### PR TITLE
Implement hit test using low level primitives

### LINKED ISSUE #11351: iOS 12 Support

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Lets see if it helps to resolve https://github.com/appium/appium/issues/11351

### CODE REVIEW

File: WebDriverAgentLib/Categories/XCUIElement+FBIsVisible.m [hunk 1]
```diff
@@ -181,13 +211,24 @@ - (BOOL)fb_isVisible
     // However, upside-down case cannot be covered this way, which is not important for Appium
     midPoint = FBInvertPointForApplication(midPoint, appFrame.size, FBApplication.fb_activeApplication.interfaceOrientation);
   }
-  XCElementSnapshot *hitElement = [self hitTest:midPoint];
-  if (nil != hitElement && (self == hitElement || [ancestors containsObject:hitElement])) {
-    return [self fb_cacheVisibilityWithValue:YES forAncestors:ancestors];
+  XCAccessibilityElement *hitElement = [self elementAtPoint:midPoint];
```
Comment (imurchie) [inline WebDriverAgentLib/Categories/XCUIElement+FBIsVisible.m:214]: In my tests just getting rid of the `hitElement` logic here entirely fixes the problem and all the rest of the tests work.

**Review by imurchie — COMMENTED** [review]
I'll test this out.

File: WebDriverAgentLib/Categories/XCUIElement+FBIsVisible.m (hunk 1 above)
Comment (mykola-mokhnach) [reply WebDriverAgentLib/Categories/XCUIElement+FBIsVisible.m:214]: visibility detection was always a pain because there are many corner cases. Perhaps, these cases just don't popup in that set of tests :(

Comment (imurchie) [reply WebDriverAgentLib/Categories/XCUIElement+FBIsVisible.m:214]: Yup. Hopefully this works.

**Review by imurchie — APPROVED** [review]
All seems good locally on iOS 12.
